### PR TITLE
[16.0][ADD] product_margin_classification: Theoretical price actions on product list views

### PR DESCRIPTION
**Pull request by carmenbianca**
A fairly straightforward feature implementation.

Internal task: https://gestion.coopiteasy.be/web#id=11378&action=479&model=project.task&view_type=form&menu_id=536

### CODE REVIEW

File: product_margin_classification/views/view_product_product.xml [hunk 1]
```diff
@@ -129,4 +129,33 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
         </field>
     </record>
 
+    <record id="action_apply_theoretical_price" model="ir.actions.server">
+        <field name="name">Apply Theoretical Price</field>
+        <field name="groups_id" eval="[(4, ref('base.group_user'))]"/>
```
Comment (carmenbianca) [inline product_margin_classification/views/view_product_product.xml]: Here and elsewhere, I am not certain this is the correct value for groups_id.

File: product_margin_classification/models/product_product.py [hunk 2]
```diff
@@ -108,3 +108,18 @@ def _get_margin_info(
...
+    def _apply_theoretical_price(self, state_list):
+        products = self.filtered(
+            lambda x: x.margin_state in state_list
+        )
+        products.use_theoretical_price()
+
+    def apply_theoretical_price(self):
+        self._apply_theoretical_price(["too_cheap", "too_expensive"])
+
+    def action_apply_theoretical_price_too_cheap(self):
```
Comment (legalsylvain) [inline product_margin_classification/models/product_product.py]:
```suggestion
    def apply_theoretical_price_too_cheap(self):
```

File: product_margin_classification/models/product_product.py [hunk 3]
```diff
@@ -108,3 +108,18 @@ def _get_margin_info(
...
+        self._apply_theoretical_price(["too_cheap"])
+
+    def action_apply_theoretical_price_too_expensive(self):
```
Comment (legalsylvain) [inline product_margin_classification/models/product_product.py]:
```suggestion
    def apply_theoretical_price_too_expensive(self):
```

File: product_margin_classification/views/view_product_product.xml (hunk 1 above)
Comment (legalsylvain) [reply product_margin_classification/views/view_product_product.xml]: the function will change the sale price for many products. 
it should be available for people that can do that operation for a product.

**Review by legalsylvain — COMMENTED** [review]
Some fixes inline.

Could you add simple test for new functions ? 

Otherwise, LGTM. thanks !

File: product_margin_classification/views/view_product_product.xml (hunk 1 above)
Comment (carmenbianca) [reply product_margin_classification/views/view_product_product.xml]: Changed this to `base.group_system`, which appears to be the group required for writing to product.product.

**Comment by carmenbianca** [comment]
Added tests for the added actions.

Did not add tests for directly executing the methods. It would be a bit of a test-the-same-thing-twice affair. But if needed, I can add them for completeness' sake.

File: product_margin_classification/views/view_product_product.xml (hunk 1 above)
Comment (legalsylvain) [reply product_margin_classification/views/view_product_product.xml]: That's a problem. base.group_system is erp manager. The feature will be hidden to all the users. To fix it : 
- set ``base.group_user``

(Core code : 
```access_product_product_employee,product.product employee,model_product_product,base.group_user,1,0,0,0```)

so the feature will be visible for all the user that can see products.
Then an  error will be raised by ORM, if user can not write on products.

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Thanks. A remark inline regarding incorrect group. otherwise, LGTM !

File: product_margin_classification/views/view_product_product.xml (hunk 1 above)
Comment (carmenbianca) [reply product_margin_classification/views/view_product_product.xml]: reverted, fixed :+1:

**Pull request merged**